frontend/src/metabase/query_builder/components/view/ViewHeader.jsx

```diff
@@ -104,23 +111,17 @@ export class ViewTitleHeader extends React.Component {
       >
         {isSaved ? (
           <div>
-            <div className="flex align-center">
-              <ViewHeading className="mr1">
-                {question.displayName()}
-              </ViewHeading>
-              {description && (
-                <Icon
-                  name="info"
-                  className="text-light mx1 cursor-pointer text-brand-hover"
-                  size={18}
-                  tooltip={description}
-                />
-              )}
-              <QuestionEntityMenu
+            <SavedQuestionHeaderButtonContainer>
+              <SavedQuestionHeaderButton
                 question={question}
-                onOpenModal={onOpenModal}
+                active={isShowingQuestionDetailsSidebar}
+                onClick={() =>
+                  isShowingQuestionDetailsSidebar
+                    ? onCloseQuestionDetails()
+                    : onOpenQuestionDetails()
+                }
```

Comment:
nip: we can just pass a function reference and avoid creating a new function causing extra re-renderings

```js
onClick={
  isShowingQuestionDetailsSidebar
    ? onCloseQuestionDetails
    : onOpenQuestionDetails
}
```

Reply:
good catch